packages/sw-build/src/lib/inject-manifest.js

```diff
@@ -0,0 +1,87 @@
+const fs = require('fs');
+const path = require('path');
+const mkdirp = require('mkdirp');
+
+const getFileManifestEntries = require('./get-file-manifest-entries');
+const errors = require('./errors');
+
+/**
+ * @example <caption>Generate a build manifest of static assets, which could
+ * then be used with a service worker.</caption>
+ * const swBuild = require('sw-build');
+ *
+ * swBuild.injectManifest({
+ *   dest: './build/'
+ *   rootDirectory: './build/',
+ *   staticFileGlobs: ['**\/*.{html,js,css}'],
+ *   globIgnores: ['admin.html'],
+ *   swFile: 'sw.js'
+ * })
+ * .then(() => {
+ *   console.log('Build Manifest generated.');
+ * });
+ *
+ * This method will generate a file manifest that can be used in a service
```

Comment:
"This method will modify an existing service worker file by finding `cacheRevisionedAssets([])` and replacing the empty `[]` array with an up to date precache manifest. This will allow the service worker to maintain a cache of local assets that will be available offline."